backend/product_management/context/controllers/responses/SneakerResponder.go

```diff
@@ -121,3 +122,50 @@ func SendSneakersByPagination(c *fiber.Ctx) error {
 		Page:     pageInt,
 	})
 }
+
+
+func SendRelatedProductsByTags(c *fiber.Ctx) error {
+	productID := c.Params("id")
+
+	var currentProduct models.Sneaker
+	objectID, err := primitive.ObjectIDFromHex(productID)
+	if err != nil {
+		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{"error": "Invalid product ID"})
+	}
+
+	filter := bson.D{{Key: "_id", Value: objectID}}
+	err = database.SneakerCollection.FindOne(context.TODO(), filter).Decode(&currentProduct)
+	if err != nil {
+		return c.Status(fiber.StatusNotFound).JSON(fiber.Map{"error": "Product not found"})
+	}
+
+	var relatedProducts []models.Sneaker
+
+	filter = bson.D{
+		{"tags", bson.D{{"$in", currentProduct.Tags[0]}}},
+		{"tags", bson.D{{"$in", currentProduct.Tags[1]}}},
+		{"_id", bson.D{{"$ne", currentProduct.ID}}},
+	}
+
+    options := options.Find().SetLimit(10)
+
+	cursor, err := database.SneakerCollection.Find(context.TODO(), filter, options)
+	if err != nil {
+		return c.Status(500).JSON(fiber.Map{"error": err.Error()})
+	}
+	defer cursor.Close(context.TODO())
+
+	for cursor.Next(context.TODO()) {
+		var product models.Sneaker
+		if err := cursor.Decode(&product); err != nil {
+			return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{"error": err.Error()})
+		}
+		relatedProducts = append(relatedProducts, product)
+	}
+
+	if len(relatedProducts) == 0 {
+		return c.Status(fiber.StatusNotFound).JSON(fiber.Map{"error": "No related products were found."})
+	}
```

Comment:
What if you do the validation with `cursor.TryNext` before the for loop in the line 158?

Reply:
Done